packages/core/src/components/containers/refine/index.tsx

```diff
@@ -152,14 +153,21 @@ export const Refine: React.FC<RefineProps> = ({
 
     const { RouterComponent } = routerProvider;
 
+    let myDataProvider;
+    if (!dataProvider.hasOwnProperty("defaultProvider")) {
```

Comment:
```suggestion
    if (typeof dataProvider !== "function") {
```
What if the user doesn't give it with the name "defaultProvider"? Should we look at it as a function?